swing/src/net/sf/openrocket/utils/SimulationTableToCSVFileExporter.java

```diff
@@ -0,0 +1,235 @@
+import net.sf.openrocket.gui.dialogs.optimization.GeneralOptimizationDialog;
+import net.sf.openrocket.gui.util.FileHelper;
+import net.sf.openrocket.gui.util.SwingPreferences;
+import net.sf.openrocket.l10n.Translator;
+import net.sf.openrocket.startup.Application;
+import net.sf.openrocket.startup.Preferences;
+
+public class SimulationTableToCSVFileExporter {
+	private final OpenRocketDocument document;
+	private final JTable simulationTable;
+	private final ColumnTableModel simulationTableModel;
+	private final HashMap<String, String> valueColumnToUnitString = new HashMap<>();
+
+	private static final Translator trans = Application.getTranslator();
+	private static final String SPACE = "SPACE";
+	private static final String TAB = "TAB";
+
+	public SimulationTableToCSVFileExporter (OpenRocketDocument document,
+			JTable simulationTable,
+			ColumnTableModel simulationTableModel			
+			) 
+	{
+		this.document = document;
+		this.simulationTable = simulationTable;
+		this.simulationTableModel = simulationTableModel;
+	}
+
+	/**
+	 * Means by which the CSV export will clean up units on the values and
+	 * describe them on the header fields instead.
+	 */
+	private void populateColumnNameToUnitsHashTable() {
+		if (null == simulationTableModel) {
+			return;
+		}
+		valueColumnToUnitString.clear(); // necessary if units changed during session
+		for (int i=0; i<simulationTableModel.getColumnCount(); i++) {
+			Column c = simulationTableModel.getColumn(i);
+			if (c instanceof ValueColumn) {
+				// only value columns seem to have units that are not zero length strings... these are
+				// the ones we actually want in our lookup table.
+				valueColumnToUnitString.put(c.toString(), c.getUnits().getDefaultUnit().getUnit());
+			}
+			
+		}
+	}
+	/**
+	 * Dump data from sim table to file for run simulations
+	 * @param data The csv data as one string block.
+	 * @param csvFile The file to dump the data to.
+	 */
+	private void dumpDataToFile(String data, File csvFile) {
+		BufferedWriter bufferedWriter = null;
+		try {
+			csvFile.createNewFile();
+			bufferedWriter = new BufferedWriter(new FileWriter(csvFile));
+			bufferedWriter.write(data);
+		} catch (FileNotFoundException e) {
+			String msg = e.getMessage();
+			JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+		} catch (IOException e) {
+			String msg = e.getMessage();
+			JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+		} finally {
+			if (bufferedWriter != null) {
+				try {
+					bufferedWriter.close();
+				} catch (IOException e) {
+					String msg = e.getMessage();
+					JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+				}
+			}
+		}
+	}
+
+	private JFileChooser setUpFileChooser() {
+		JFileChooser fch = new JFileChooser();
+		String saveDialogTitle = trans.get("simpanel.pop.export_to_csv.save.dialog.title");
+		String saveButtonText = trans.get("simpanel.pop.export_to_csv.save.button.text");
```

Comment:
There is no real added value to defining a custom save button text (and tooltip), I would remove it.